Develop

Gomina opened the pull request.

Gomina committed:
add allure report

Gomina committed:
Commit 1
Добавлены файлы с тестами

DolgoarshinnykhLG commented on helper.py:
```diff
@@ -0,0 +1,105 @@
+    @contextmanager
+    def register_new_courier_and_return_login_password():
+        # метод генерирует строку, состоящую только из букв нижнего регистра, в качестве параметра передаём длину строки
+        def generate_random_string(length):
+            letters = string.ascii_lowercase
+            random_string = ''.join(random.choice(letters) for i in range(length))
+            return random_string
```
Можно лучше: этот метод лучше вынести из этого класса и оставить функцией модуля, т.к. данный метод является именно вспомогательным для CouriersMethods

DolgoarshinnykhLG commented on helper.py:
```diff
@@ -0,0 +1,105 @@
+        # если регистрация прошла успешно (код ответа 201), добавляем в список логин и пароль курьера
+        if response.status_code == 201:
+            login_pass.extend([login, password, first_name])
+            print(f"Успешная регистрация! Логин: {login}, Пароль: {password}, Имя: {first_name}")
```
Нужно исправить здесь и далее: от любых принтов в проекте лучше избавиться. Если есть желание логировать что-то, то лучше использовать пакет logging

DolgoarshinnykhLG commented on helper.py:
```diff
@@ -0,0 +1,105 @@
+    # если регистрация не удалась, возвращает пустой список
+    @allure.step('метод создает курьера с случайными логином, паролем, именем. После теста удаляет курьера')
+    @contextmanager
+    def register_new_courier_and_return_login_password():
```
Нужно исправить: метод получился слишком объёмным и очень схож с given_register_new_courier по логике. Лучше разделить его на создание и удаление, а их вызовы которые не являются шагом теста (например, удаление курьера, а в некоторых случаях и создание) вынести в фикстуры

DolgoarshinnykhLG commented on tests/test_couriers.py:
```diff
@@ -0,0 +1,164 @@
+        assert content.get("message") == "Этот логин уже используется. Попробуйте другой."
+        # удаление курьера
+        courier_id = couriers_methods.login_courier(TC.COURIER_1['login'], TC.COURIER_1['password'])
+        if courier_id:
+            couriers_methods.delete_courier(courier_id[0])
```
Нужно исправить здесь и далее: тесты должны заканчиваться assert-инструкцией. Удаление курьера тут является пост-условием, можно вынести в фикстуру

DolgoarshinnykhLG commented on tests/test_couriers.py:
```diff
@@ -0,0 +1,164 @@
+        couriers_methods = CouriersMethods()
+        # создание курьера
+        couriers_methods.given_register_new_courier(**TC.COURIER_1)
+        try:
```
Нужно исправить здесь и далее: Необходимо избавиться от try-except в теле теста, можно использовать pytest.raises

DolgoarshinnykhLG commented on urls.py:
```diff
@@ -0,0 +1,11 @@
+class URL:
+
+    BASE_URL = 'https://qa-scooter.praktikum-services.ru'
+    CREATE_COURIER = BASE_URL + '/api/v1/courier'
+    LOGIN_COURIER = BASE_URL + '/api/v1/courier/login'
+    DELETE_COURIER = BASE_URL + '/api/v1/courier/'
+    CREATE_ORDER = BASE_URL + '/api/v1/orders'
+    ORDER_LIST = BASE_URL + '/api/v1/orders'
```
Можно лучше: url-адреса можно конкатенировать с базовым (BASE_URL), так будет проще поддерживать несколько стендов

Gomina committed:
Commit 2
Url-адреса конкатенироватys с базовым (BASE_URL)

Gomina committed:
Commit 3
Из тестов удалены try-except

Gomina committed:
Commit 4
Удаление курьера убрано в фикстуру

Gomina committed:
Commit 5
Метод register_new_courier_and_return_login_password разбит на более мелкие методы и фикстуру

Gomina committed:
Commit 6
Print удалены из проекта

Gomina committed:
add allure report

DolgoarshinnykhLG commented on tests/test_couriers.py:
```diff
@@ -0,0 +1,135 @@
+    @allure.title('Тест на успешное создание рандомного курьера')
+    def test_creating_courier_success(self, temporary_courier):
+        # рандомный курьер из фикстуры
+        result = temporary_courier
```
Необходимо исправить: создание курьера - это шаг сценария теста, а сейчас он фактически создаётся в предусловии. Необходимо для этого теста перенести создание курьера в шаги теста

Gomina committed:
Commit 7
Добавлен шаг "Создание курьера" в тест test_creating_courier_success
